src/test/url-loader/package-url-resolver_test.ts

```diff
@@ -152,8 +152,46 @@ suite('PackageUrlResolver', function() {
       assert.equal(
           resolver.resolve(fileRelativeUrl`%><><%=`, packageRoot), undefined);
     });
-  });
 
+    test('resolves a relative URL containing querystring and fragment', () => {
+      assert.equal(
+          resolver.resolve(
+              fileRelativeUrl`foo.html#bat`, rootedFileUrl`1/2/foo.html?bar`),
+          rootedFileUrl`1/2/foo.html#bat`);
+      assert.equal(
+          resolver.resolve(fileRelativeUrl`foo.html?baz#bat`, packageRoot),
+          rootedFileUrl`1/2/foo.html?baz#bat`);
+      assert.equal(
+          resolver.resolve(
+              fileRelativeUrl`foo.html?baz#bat`, rootedFileUrl`1/2/bar/baz/`),
+          rootedFileUrl`1/2/bar/baz/foo.html?baz#bat`);
+    });
+
+    test('resolves a URL with no pathname', () => {
+      assert.equal(
+          resolver.resolve(fileRelativeUrl``, rootedFileUrl`1/2/foo.html`),
+          rootedFileUrl`1/2/foo.html`);
+      assert.equal(
+          resolver.resolve(
+              fileRelativeUrl``, rootedFileUrl`1/2/foo.html?baz#bat`),
+          rootedFileUrl`1/2/foo.html?baz`);
+      assert.equal(
+          resolver.resolve(fileRelativeUrl`#buz`, rootedFileUrl`1/2/foo.html`),
+          rootedFileUrl`1/2/foo.html#buz`);
+      assert.equal(
+          resolver.resolve(
+              fileRelativeUrl`#buz`, rootedFileUrl`1/2/foo.html?baz#bat`),
+          rootedFileUrl`1/2/foo.html?baz#buz`);
+      assert.equal(
+          resolver.resolve(
+              fileRelativeUrl`?fiz#buz`, rootedFileUrl`1/2/foo.html`),
+          rootedFileUrl`1/2/foo.html?fiz#buz`);
+      assert.equal(
+          resolver.resolve(
+              fileRelativeUrl`?fiz#buz`, rootedFileUrl`1/2/foo.html?bar`),
+          rootedFileUrl`1/2/foo.html?fiz#buz`);
```

Comment:
Could use a test that just a query string overrides the fragment in the base

```typescript
      assert.equal(
          resolver.resolve(
              fileRelativeUrl`?fiz`, rootedFileUrl`1/2/foo.html?bar#buz`),
          rootedFileUrl`1/2/foo.html?fiz`);
```

Reply:
Added.